### PR TITLE
Add animal fixes

### PR DESCRIPTION
**Pull request by SayakaOno**
**Description**

Address Duncan's [comment](https://github.com/LiteFarmOrg/LiteFarm/pull/3397#pullrequestreview-2270361189) in #3397.  There might be better solutions and it might require adjustments. Please feel free to update or close as needed. Thank you! 🙏 

- Filter out animal uses that are not for the currently selected animal type when coming back to the details view.
- Adjust summary image width (The image might look too small at some breatkpoints.)

Jira link: N/A

**Type of change**

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**How Has This Been Tested?**

- [ ] Passes test case
- [x] UI components visually reviewed on desktop view
- [ ] UI components visually reviewed on mobile view
- [ ] Other (please explain)

**Checklist:**

- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] The precommit and linting ran successfully
- [ ] I have added or updated language tags for text that's part of the UI
- [ ] I have added "MISSING" for all new language tags to languages I don't speak
- [ ] I have added [the GNU General Public License](https://lite-farm.atlassian.net/l/cp/BT0Dd7WW) to all new files

### CODE REVIEW

**Comment by kathyavini** [comment]
Thank you @SayakaOno for working on this before you leave (and thanks for making a separate PR -- that definitely helps with the review!)

Sorry, I should have read Duncan's comments closer as I didn't realize the uses bug was from my hook and not related to LF-4166 at all. Thank you for suggesting a fix 🙏 

Have a great vacation!! ❤️ 🛫  See you in October

**Comment by SayakaOno** [comment]
No worries!!
Thank you Joyce! ❤️ Itte kimasu 🛫 See you in October!

**Review by kathyavini — APPROVED** [review]
I had never given any thought to the idea of filtering the user's already input uses based on the current type, but I think this is the most logical approach. As it had not been discussed before I think we could give it a quick check with Loïc tomorrow, but I'm more than happy to merge as is first.

If Loïc likes the behaviour, we might also need it in the component anyway for readonly/edit, but I'm happy to have it reside here until that is built out!

To sum up a Slack conversation, I believe @Duncan-Brain actually wasn't requesting this adjust of previous selected items, but instead was just seeing the core filtering logic not working at some point on the other branch. But this is a real use case worth addressing, I think.

**Comment by Duncan-Brain** [comment]
It seems weird to keep any of the use data if switching types, maybe in the future we can just clear it. 

I definitely just meant it was not filtering! haha But this makes sense too.
